tests/pyspark/test_pyspark_model.py

```diff
@@ -300,19 +300,34 @@ def function_expected():
     )
 
 
-def test_optional_column() -> None:
+def test_optional_column(spark) -> None:
     """Test that optional columns are not required."""
 
     class Schema(DataFrameModel):  # pylint:disable=missing-class-docstring
-        a: Optional[str]
-        b: Optional[str] = pa.Field(eq="b")
-        c: Optional[str]  # test pandera.typing alias
+        a: str
+        b: int
+        c: Optional[str] = pa.Field(eq="Food")
+        d: Optional[T.StringType] = pa.Field(eq="Food")
 
     schema = Schema.to_schema()
-    assert not schema.columns["a"].required
-    assert not schema.columns["b"].required
+    assert schema.columns["a"].required
+    assert schema.columns["b"].required
```

Comment:
Changing current test case to check mandatory columns too.

Reply:
rather than changing the existing test, I recommend adding a new test for a mixed case. This test is named specifically for `optional`.

Reply:
I changed the approach: deepening the scope of this test to not just check if Optional fields are valid in the schema (`.required`) but also to check if it's processable through a `.validate()` call.